src/components/App.js

```diff
@@ -3,12 +3,36 @@ import React from 'react';
 import ButtonPanel from './ButtonPanel';
 import Display from './Display';
 import '../assets/main.css';
+import calculate from '../logic/calculate';
 
-const App = () => (
-  <div id="app-container">
-    <Display />
-    <ButtonPanel />
-  </div>
-);
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+  }
+
+  handleClick = buttonName => {
+    const { ...copyState } = this.state;
+    const { ...calcResult } = calculate(copyState, buttonName);
+    this.setState(calcResult);
```

Comment:
You can use React's `prevState` for this. You could read more about it [here](https://stackoverflow.com/questions/54807454/what-is-prevstate-in-reactjs). This is completely optional, just pointing it out for future reference.

```suggestion
    this.setState(prevState => ({
        calculate(prevState, buttonName)
    }))
}));
```